.changeset/poor-avocados-act.md

```diff
@@ -0,0 +1,30 @@
+---
+'merchant-center-application-template-starter': major
+'@commercetools-frontend/application-config': major
+'@commercetools-frontend/application-shell': major
+'@commercetools-frontend/application-shell-connectors': major
+'@commercetools-frontend/mc-dev-authentication': major
+'@commercetools-frontend/mc-html-template': major
+'@commercetools-frontend/mc-scripts': major
+'playground': major
+'@commercetools-local/visual-testing-app': major
+'@commercetools-website/custom-applications': major
+---
+
+Remove the CLI flag `--use-local-assets`. The default behavior of `mc-scripts compile-html` now is to compile the assets locally, which is the only reasonable thing to do.
+
+Furthermore, the `@commercetools-frontend/mc-http-server` package has been deprecated and won't be published anymore.
+With the `compile-html` command there is no need to have a pre-configured HTTP server anymore.
+
+When running the `mc-scripts compile-html` command, the `index.html` is compiled for production usage and it lives in the `public` folder, together with the other static assets. This is all you need to deploy your application.
+You can decide to [deploy the Custom Application statically to one of the popular cloud providers](https://docs.commercetools.com/custom-applications/deployment/compiling-a-custom-application#deployment), or serve the files on your own using a static server.
+
+For example, to run locally the Custom Application using the production bundles:
+
+```console
+NODE_ENV=production MC_APP_ENV=development dotenv -- \
+  mc-scripts compile-html \
+  --transformer @commercetools-frontend/mc-dev-authentication/transformer-local.js
+
+mc-scripts serve
```

Comment:
I forgot that we need to serve the files as a SPA.
The `serve` package has a `--single` option that does that but then we loose the routes for the login/logout pages.

Therefore I decided to add a new `serve` command to `mc-scripts`, so that we can customize the rewrite routes.
A `serve` command is something that is also used in other tools, like Gatsby.

In our case the `mc-scripts serve` command starts an HTTP server to serve the files in the `public` folder. The main use case for this is to start the Custom Application locally to test the production bundles.